src/sidebar/components/annotation-viewer-content.js

```diff
@@ -1,77 +1,119 @@
-      annot = annot_;
-      return api.search({ references: annot.id });
-    })
-    .then(function (searchResult) {
-      return [annot].concat(searchResult.rows);
-    });
-}
-
-// @ngInject
-function AnnotationViewerContentController(
-  store,
+function AnnotationViewerContent({
   api,
-  rootThread,
+  rootThread: rootThreadService,
   streamer,
-  streamFilter
-) {
-  store.clearAnnotations();
+  streamFilter,
+}) {
+  const addAnnotations = useStore(store => store.addAnnotations);
+  const annotationId = useStore(store => store.routeParams().id);
+  const clearAnnotations = useStore(store => store.clearAnnotations);
+  const highlightAnnotations = useStore(store => store.highlightAnnotations);
+  const rootThread = useStore(store =>
+    rootThreadService.thread(store.getState())
+  );
+  const setCollapsed = useStore(store => store.setCollapsed);
 
-  const annotationId = store.routeParams().id;
+  useEffect(() => {
+    clearAnnotations();
 
-  this.rootThread = () => rootThread.thread(store.getState());
+    // TODO - Handle exceptions during the `fetchThread` call.
+    fetchThread(api, annotationId).then(annots => {
+      addAnnotations(annots);
 
-  this.setCollapsed = function (id, collapsed) {
-    store.setCollapsed(id, collapsed);
-  };
+      const topLevelAnnot = annots.filter(
+        ann => (ann.references || []).length === 0
+      )[0];
 
-  this.ready = fetchThread(api, annotationId).then(function (annots) {
-    store.addAnnotations(annots);
-
-    const topLevelAnnot = annots.filter(function (annot) {
-      return (annot.references || []).length === 0;
-    })[0];
+      if (!topLevelAnnot) {
+        // `annotationId` refers to a reply. We were able to fetch the reply
+        // but not the top-level annotation at the top of the thread.
+        //
+        // We need to decide what what be the most useful behavior in this case
+        // and implement it.
+        /* istanbul ignore next */
+        return;
+      }
 
-    if (!topLevelAnnot) {
-      return;
-    }
+      streamFilter
+        .addClause('/references', 'one_of', topLevelAnnot.id, true)
+        .addClause('/id', 'equals', topLevelAnnot.id, true);
+      streamer.setConfig('filter', { filter: streamFilter.getFilter() });
+      streamer.connect();
 
-    streamFilter
-      .addClause('/references', 'one_of', topLevelAnnot.id, true)
-      .addClause('/id', 'equals', topLevelAnnot.id, true);
-    streamer.setConfig('filter', { filter: streamFilter.getFilter() });
-    streamer.connect();
+      annots.forEach(annot => setCollapsed(annot.id, false));
 
-    annots.forEach(function (annot) {
-      store.setCollapsed(annot.id, false);
+      if (topLevelAnnot.id !== annotationId) {
+        // FIXME - This should show a visual indication of which reply the
+        // annotation ID in the URL refers to. That isn't working.
+        highlightAnnotations([annotationId]);
+      }
     });
+  }, [
+    addAnnotations,
```

Comment:
All except one of these dependencies is actually static, in that it should never change in the real app. Unfortunately the linter doesn't know that. Only the `annotationId` is dynamic. I will demarcate them to make this clearer.

> is there ever any benefit into factoring into multiple useEffect hooks—not for any functional difference, but because it's easier to maintain and keep the functionality bundled up?

My conceptual understanding is that the purpose of an effect is to synchronize some output with its inputs. In this case, to synchronize the set of loaded annotations with the annotation ID in the URL. It makes sense to split up effects if logically there are different _outputs_ to synchronize which have different dependencies. This avoids unnecessary work in running effects which won't produce any change. In this case, all of the effects depend on the one input - `annotationId`.